traffic_ops/v3-client/asn.go

```diff
@@ -68,24 +68,40 @@ func (to *Session) UpdateASNByID(id int, entity tc.ASN) (tc.Alerts, ReqInf, erro
 }
 
 // Returns a list of ASNs
-func (to *Session) GetASNs() ([]tc.ASN, ReqInf, error) {
-	resp, remoteAddr, err := to.request(http.MethodGet, API_ASNS, nil, nil)
+func (to *Session) GetASNsWithHeader(header http.Header) ([]tc.ASN, ReqInf, error) {
```

Comment:
I think since we're updating the ASNs client methods here, we should make it take a `*url.Values` as the first parameter, which allows the user to pass the ID or ASN in the request. Then we no longer need `GetASNByIDWithHeader` and `GetASNByASNWithHeader`. I believe that is the pattern we're trying to move towards these days.

Reply:
Ah ok.. I had no idea, but will update accordingly.